app/lib/proof_provider/keybase.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+class ProofProvider::Keybase
+  BASE_URL = 'https://keybase.io'
+
+  class Error < StandardError; end
+
+  class ExpectedProofLiveError < Error; end
+
+  class UnexpectedResponseError < Error; end
```

Comment:
One practice I've taken up is having a global exception directory (and error): `app/errors` and `app/exceptions`, so that errors can be shared and to deliniate between business problems and developer problems (example: `ArgumentError` vs `NotEnoughFundsException`

Reply:
We do have an `app/lib/errors.rb` file with that sort of thing. My intention here is to contain Keybase-specific errors in the Keybase-specific area of the code--the Rails controllers/models should not receive these exceptions.